src/main/kotlin/org/rust/cargo/project/model/impl/CargoProjectImpl.kt

```diff
@@ -222,20 +223,31 @@ open class CargoProjectsServiceImpl(
         modifyProjects { doRefresh(project, it) }
 
     override fun discoverAndRefresh(): CompletableFuture<out List<CargoProject>> {
-        val guessManifest = suggestManifests().firstOrNull()
-            ?: return CompletableFuture.completedFuture(projects.currentState)
+        val guessManifests = suggestManifests()
 
         return modifyProjects { projects ->
-            if (hasAtLeastOneValidProject(projects)) return@modifyProjects CompletableFuture.completedFuture(projects)
-            doRefresh(project, listOf(CargoProjectImpl(guessManifest.pathAsPath, this)))
+            doRefresh(project, guessManifests.map { CargoProjectImpl(it.pathAsPath, this) }.toList())
         }
     }
 
-    override fun suggestManifests(): Sequence<VirtualFile> =
-        project.modules
-            .asSequence()
-            .flatMap { ModuleRootManager.getInstance(it).contentRoots.asSequence() }
-            .mapNotNull { it.findChild(CargoConstants.MANIFEST_FILE) }
+    override fun suggestManifests(): Sequence<VirtualFile> {
```

Comment:
Detecting Cargo projects recursively generally makes the plugin more powerful, as it avoids the user having to repeatedly manually "Attach Cargo Project". This is especially true in a Bazel project where the `Cargo.toml` files are hidden away in the Bazel sandbox, out of sight.